tests/utils/test_Vec2.cpp

```diff
@@ -1,333 +1,436 @@
+    ASSERT_EQ(x, vecXXXX.z);
+    ASSERT_EQ(x, vecXXXX.w);
+    ASSERT_EQ(x, vecXXXY.x);
+    ASSERT_EQ(x, vecXXXY.y);
+    ASSERT_EQ(x, vecXXXY.z);
+    ASSERT_EQ(y, vecXXXY.w);
+    ASSERT_EQ(x, vecXXYX.x);
+    ASSERT_EQ(x, vecXXYX.y);
+    ASSERT_EQ(y, vecXXYX.z);
+    ASSERT_EQ(x, vecXXYX.w);
+    ASSERT_EQ(x, vecXXYY.x);
+    ASSERT_EQ(x, vecXXYY.y);
+    ASSERT_EQ(y, vecXXYY.z);
+    ASSERT_EQ(y, vecXXYY.w);
+    ASSERT_EQ(x, vecXYXX.x);
+    ASSERT_EQ(y, vecXYXX.y);
+    ASSERT_EQ(x, vecXYXX.z);
+    ASSERT_EQ(x, vecXYXX.w);
+    ASSERT_EQ(x, vecXYXY.x);
+    ASSERT_EQ(y, vecXYXY.y);
+    ASSERT_EQ(x, vecXYXY.z);
+    ASSERT_EQ(y, vecXYXY.w);
+    ASSERT_EQ(x, vecXYYX.x);
+    ASSERT_EQ(y, vecXYYX.y);
+    ASSERT_EQ(y, vecXYYX.z);
+    ASSERT_EQ(x, vecXYYX.w);
+    ASSERT_EQ(x, vecXYYY.x);
+    ASSERT_EQ(y, vecXYYY.y);
+    ASSERT_EQ(y, vecXYYY.z);
+    ASSERT_EQ(y, vecXYYY.w);
+    ASSERT_EQ(y, vecYXXX.x);
+    ASSERT_EQ(x, vecYXXX.y);
+    ASSERT_EQ(x, vecYXXX.z);
+    ASSERT_EQ(x, vecYXXX.w);
+    ASSERT_EQ(y, vecYXXY.x);
+    ASSERT_EQ(x, vecYXXY.y);
+    ASSERT_EQ(x, vecYXXY.z);
+    ASSERT_EQ(y, vecYXXY.w);
+    ASSERT_EQ(y, vecYXYX.x);
+    ASSERT_EQ(x, vecYXYX.y);
+    ASSERT_EQ(y, vecYXYX.z);
+    ASSERT_EQ(x, vecYXYX.w);
+    ASSERT_EQ(y, vecYXYY.x);
+    ASSERT_EQ(x, vecYXYY.y);
+    ASSERT_EQ(y, vecYXYY.z);
+    ASSERT_EQ(y, vecYXYY.w);
+    ASSERT_EQ(y, vecYYXX.x);
+    ASSERT_EQ(y, vecYYXX.y);
+    ASSERT_EQ(x, vecYYXX.z);
+    ASSERT_EQ(x, vecYYXX.w);
+    ASSERT_EQ(y, vecYYXY.x);
+    ASSERT_EQ(y, vecYYXY.y);
+    ASSERT_EQ(x, vecYYXY.z);
+    ASSERT_EQ(y, vecYYXY.w);
+    ASSERT_EQ(y, vecYYYX.x);
+    ASSERT_EQ(y, vecYYYX.y);
+    ASSERT_EQ(y, vecYYYX.z);
+    ASSERT_EQ(x, vecYYYX.w);
+    ASSERT_EQ(y, vecYYYY.x);
+    ASSERT_EQ(y, vecYYYY.y);
+    ASSERT_EQ(y, vecYYYY.z);
+    ASSERT_EQ(y, vecYYYY.w);
 }
 
-UTEST(test_Vec2, GetString)
+UTEST(test_Vec2, TestToString)
 {
-    String expected = "5.00,10.00";
+    String expected = "1.00,2.00";
+
+    Vec2 vec = {1.f, 2.f};
 
-    Vec2 vec0 = {5, 10};
-    String result = vec0.ToString();
+    String result = ToString(vec);
 
-    ASSERT_STREQ(expected.Str(), result.Str());
+    ASSERT_STREQ(result.Str(), expected.Str());
 }
 
-UTEST(test_Vec2, GetVectorLength)
+UTEST(test_Vec2, TestFromString)
 {
-    Vec2 vec0 = {2, 3};
+    Vec2 expected = {1.f, 2.f};
 
-    float length = vec0.Length();
+    String inputString = "1.00,2.00";
 
-    ASSERT_EQ(3.60555124f, length);
+    Vec2 result {};
+
+    ASSERT_TRUE(FromString(result, inputString));
+
+    ASSERT_EQ_VEC2(result, expected);
 }
 
-UTEST(test_Vec2, NormaliseVector)
+UTEST(test_Vec2, TestFuzzyEquals)
 {
-    Vec2 vec0 = {2, 3};
+    Vec2 vecA = {1.f, 2.f};
 
-    Vec2 norm = vec0.Normalise();
+    Vec2 vecB = {0.9999999999999912, 2.0000000000000023f};
 
-    ASSERT_TRUE((norm.x - 0.554775417f) < std::numeric_limits<float>::epsilon());
-    ASSERT_TRUE((norm.y - 0.832163095f) < std::numeric_limits<float>::epsilon());
```

Comment:
Again, should test alternate case on boundary

Reply:
Changed.